openslides_backend/presenter/presenter.py

```diff
@@ -120,16 +117,13 @@ def parse_presentations(self, payload: Payload) -> PresenterResponse:
         )
         response = []
         for presentation_blob in payload:
-            # user_id = presentation_blob["user_id"]
-            if presentation_blob["presentation"] == "dummy":
-                response.append(self.dummy_presentation())
+            Presenter = presenters.get(presentation_blob["presenter"])
+            if Presenter is not None:
+                p = Presenter()
+                response.append(p.data)
             else:
                 raise PresenterException(
```

Comment:
Hi, one note about the `PresenterException` (see https://github.com/OpenSlides/OpenSlides/blob/openslides4-dev/docs/interfaces/presenter-service.txt#L24):

You must differentiate between a server-error vs. client-error (analog to HTTP 400 vs 500). A presenter may raise an exception (you have to decide if it is the `PresenterException` or something else) on user errors. E.g. a number was expected as the payload but something else was given. This must result in a HTTP 200 response with `[{error: "a number was expected"}]` as the payload. A HTTP 500 can be raised, if e.g. a dependend service failed.

For non-existent presenters (e.g. the clients request the presenter `not-existent`) there are two possibles: Either use the error response as stated above. This allowes for the execution of other presenters, if multiple were requested. The other option is to check for validity of all presenters before executing them. Then you may raise a 400-error with a hint that some presenters are unknown.

Reply:
Ah I am not sure about this. See actions module for the same case. The module may always use Python exceptions maybe with different code names so that the http component may decide whether to send HTTP 4xx or 5xx. If a user sends a bad payload he should get HTTP 4xx and not HTTP 200 with error message. in my opinion. But that is still discussable. For the first implementation I would recommend the way we use in action module so we have a first consistent implementation here.

Reply:
Regarding HTTP-Status-Codes I share the view of @normanjaeckel on this.
There is a relatively clear semantics on this: 

`HTTP 200` means `OK` 
See https://tools.ietf.org/html/rfc7231#section-6.3

`HTTP 400`: is a formally malformed request; that excludes questions of validity.
See https://tools.ietf.org/html/rfc7231#section-6.5.1 
> malformed request syntax, invalid request message framing, or deceptive request routing

`HTTP 500`: is a formally wellformed request but the server can not deal properly with the content. This would include questions on validation.

See https://tools.ietf.org/html/rfc7231#section-6.6 

I have a look at how this was done in the actions module :]

Reply:
We are using the status codes for different reasons here, since one can send multiple presenters. presenter-scoped arrors should not stop the execution of other presenters and must so reported as error dicts.

HTTP status codes, on the other hand, are request-scoped, meaning the payload of all presenters. If e.g. a non-existing presenter is specified it is a request-scoped error resulting in a 400er.

We should divide the both layers where errors can happen.

Reply:
I opt for leaving this out of this pull request. Otherwise this PR gets _feature creep_ and I start fixing too many things at once. 

Dealing with results of Actions or Presenters is part of https://github.com/OpenSlides/openslides-backend/issues/32#event-3279225558
Perhaps this is a better place to talk about this - or even a separate issue.